core/src/main/java/org/infinispan/protostream/config/impl/AnnotationAttributeConfigurationImpl.java

```diff
@@ -66,6 +74,22 @@ public Set<String> allowedValues() {
       return allowedValues;
    }
 
+   @Override
+   public boolean isAllowed(AnnotationElement.Value value) {
+      if (allowedValues != null) {
+         String v = String.valueOf(value.getValue());
+         if (allowedValues.contains(v)) {
+            return true;
+         } else if (packageName != null && v.startsWith(packageName)) {
+            return allowedValues.contains(v.substring(packageName.length() + 1));
+         } else {
+            return false;
+         }
```

Comment:
small nit if you want 
```suggestion
         return allowedValues.contains(v) ||
               packageName != null && v.startsWith(packageName) && allowedValues.contains(v.substring(packageName.length() + 1));
```